app/com/linkedin/drelephant/spark/heuristics/ConfigurationParametersHeuristic.scala

```diff
@@ -0,0 +1,485 @@
+    evaluator.sparkExecutorInstances.foreach { numExecutors =>
+      resultDetails += new HeuristicResultDetails(CURRENT_SPARK_EXECUTOR_INSTANCES, numExecutors.toString)
+    }
+    evaluator.sparkExecutorMemoryOverhead.foreach { memOverhead =>
+      resultDetails += new HeuristicResultDetails(CURRENT_SPARK_EXECUTOR_MEMORY_OVERHEAD,
+        bytesToString(memOverhead))
+    }
+    evaluator.sparkDriverMemoryOverhead.foreach { memOverhead =>
+      resultDetails += new HeuristicResultDetails(CURRENT_SPARK_DRIVER_MEMORY_OVERHEAD,
+        bytesToString(memOverhead))
+    }
+
+    resultDetails ++= Seq(
+      new HeuristicResultDetails(RECOMMENDED_SPARK_EXECUTOR_CORES,
+        evaluator.recommendedExecutorCores.toString),
+      new HeuristicResultDetails(RECOMMENDED_SPARK_EXECUTOR_MEMORY,
+        bytesToString(evaluator.recommendedExecutorMemory)),
+      new HeuristicResultDetails(RECOMMENDED_SPARK_MEMORY_FRACTION,
+        evaluator.recommendedMemoryFraction.toString),
+      new HeuristicResultDetails(RECOMMENDED_SPARK_DRIVER_CORES,
+        evaluator.recommendedDriverCores.toString),
+      new HeuristicResultDetails(RECOMMENDED_SPARK_DRIVER_MEMORY,
+        bytesToString(evaluator.recommendedDriverMemory))
+     )
+    evaluator.recommendedExecutorInstances.foreach { numExecutors =>
+      resultDetails += new HeuristicResultDetails(RECOMMENDED_SPARK_EXECUTOR_INSTANCES,
+        numExecutors.toString)
+    }
+    evaluator.recommendedExecutorMemoryOverhead.foreach { memoryOverhead =>
+      resultDetails += new HeuristicResultDetails(RECOMMENDED_SPARK_EXECUTOR_MEMORY_OVERHEAD,
+        bytesToString(memoryOverhead))
+    }
+    evaluator.recommendedDriverMemoryOverhead.foreach { memoryOverhead =>
+      resultDetails += new HeuristicResultDetails(RECOMMENDED_SPARK_DRIVER_MEMORY_OVERHEAD,
+        bytesToString(memoryOverhead))
+    }
+    if (evaluator.stageDetails.getValue.length > 0) {
+      resultDetails += evaluator.stageDetails
+    }
+
+    new HeuristicResult(
+      heuristicConfigurationData.getClassName,
+      heuristicConfigurationData.getHeuristicName,
+      evaluator.severity,
+      evaluator.score,
+      resultDetails.asJava
+    )
+  }
+}
+
+ object ConfigurationParametersHeuristic {
+   import ConfigurationHeuristicsConstants._
+
+   /**
+     * Evaluate the metrics for a given Spark application, and determine recommended configuration
+     * parameter values
+     *
+     * @param configurationParametersHeuristic configuration parameters heurisitc
+     * @param data Spark application data
+     */
+   class Evaluator(
+       configurationParametersHeuristic: ConfigurationParametersHeuristic,
+       data: SparkApplicationData) {
+     lazy val appConfigurationProperties: Map[String, String] =
+       data.appConfigurationProperties
+
+     // current configuration parameters
+     lazy val sparkExecutorMemory = JavaUtils.byteStringAsBytes(
```

Comment:
Can't we use MemoryFormatUtils#stringToBytes which is in Dr.Elephant codebase instead of using Spark utils. Utils are typically meant to be used within a project even though they are public classes.

Reply:
Looking at this again, this should be calling byteStringAsMb. Alternatively, MemoryFormatUtils#stringToBytes  could be modified to take a default unit as well.